Add images to extensions page

yscik opened the pull request:
Depends on #4197 

### Changes proposed in this Pull Request

* Add support for displaying extension images in cards in various layouts
  — Image is on top on featured & grid cards, on the left side for large lists, and no image for small lists
* Update card layout: set cards in the same row to the same height, align card action buttons to bottom
* Update large lists to left-aligned action buttons


### Testing instructions

* 


<!--
Helpful tips for screenshots:
https://en.support.wordpress.com/make-a-screenshot/
-->
### Screenshot / Video

![image](https://user-images.githubusercontent.com/176949/116155320-3d8ad980-a6ea-11eb-9701-4b46d425558d.png)

renatho commented:
Code looks good to me! I just noticed some scenarios where the image doesn't fit very well, depending on the resolution. Like this:

<img width="874" alt="Screen Shot 2021-04-29 at 10 20 13" src="https://user-images.githubusercontent.com/876340/116557573-edbc3600-a8d4-11eb-9898-54fff6728f4a.png">

Maybe we could make an image with marks on some limits, and ask a design review to add the main content of the image inside these marks. WDYT?

Also, about the large list, I think for small resolutions we could position the image on top of the card, similar to the grid.

<img width="419" alt="Screen Shot 2021-04-29 at 10 19 24" src="https://user-images.githubusercontent.com/876340/116557764-265c0f80-a8d5-11eb-9273-953f8005cd2d.png">

> When there is no image, I added a light gray background. Maybe this default should have a placeholder, like a Sensei logo?

I think something like Sensei logo inside the gray would be nice.

yscik commented:
I've played around with some options for image alignment, and `cover` still works best. I've adjusted the images a bit to have a wider background, and moved up the center for WCPC so the purchase button won't get cut off. 

Also updated large list in mobile mode to have top images.

renatho reviewed (approved):
Cool! Looks good now! =)

I wonder if we could already merge the 82-gh-wpcomvip/a8c-senseilms-com, and tweak it here before merging.

yscik commented:
We'd need to set featured images for the extensions, which will show up for everyone, so that needs to wait until release :)
But updated the code to use the new field, with the mocked URL as a fallback in dcd525c

renatho commented:
> We'd need to set featured images for the extensions, which will show up for everyone, so that needs to wait until release :)
> But updated the code to use the new field, with the mocked URL as a fallback in [dcd525c](https://github.com/Automattic/sensei/commit/dcd525c52af4ba22bfa2c934189f23ce8776316f)

Ah, I thought it was already registered as featured images there. I added it to the Trello card of the images, so we don't forget. =)
